broker-core/src/main/java/io/zeebe/broker/workflow/processor/BpmnStepProcessor.java

```diff
@@ -0,0 +1,186 @@
+import io.zeebe.broker.logstreams.processor.TypedStreamWriter;
+import io.zeebe.broker.workflow.data.WorkflowInstanceRecord;
+import io.zeebe.broker.workflow.map.ActivityInstanceMap;
+import io.zeebe.broker.workflow.map.DeployedWorkflow;
+import io.zeebe.broker.workflow.map.PayloadCache;
+import io.zeebe.broker.workflow.map.WorkflowCache;
+import io.zeebe.broker.workflow.map.WorkflowInstanceIndex;
+import io.zeebe.broker.workflow.map.WorkflowInstanceIndex.WorkflowInstance;
+import io.zeebe.broker.workflow.model.BpmnStep;
+import io.zeebe.broker.workflow.model.ExecutableFlowElement;
+import io.zeebe.broker.workflow.model.ExecutableWorkflow;
+import io.zeebe.broker.workflow.processor.activity.CreateJobHandler;
+import io.zeebe.broker.workflow.processor.activity.InputMappingHandler;
+import io.zeebe.broker.workflow.processor.activity.OutputMappingHandler;
+import io.zeebe.broker.workflow.processor.exclusivegw.ExclusiveSplitHandler;
+import io.zeebe.broker.workflow.processor.flownode.ConsumeTokenHandler;
+import io.zeebe.broker.workflow.processor.flownode.TakeSequenceFlowHandler;
+import io.zeebe.broker.workflow.processor.sequenceflow.ActivateGatewayHandler;
+import io.zeebe.broker.workflow.processor.sequenceflow.EnterIntermediateEventHandler;
+import io.zeebe.broker.workflow.processor.sequenceflow.StartActivityHandler;
+import io.zeebe.broker.workflow.processor.sequenceflow.TriggerEndEventHandler;
+import io.zeebe.logstreams.processor.EventLifecycleContext;
+import io.zeebe.protocol.intent.WorkflowInstanceIntent;
+import io.zeebe.transport.ClientResponse;
+import io.zeebe.util.sched.ActorControl;
+import io.zeebe.util.sched.future.ActorFuture;
+import io.zeebe.util.sched.future.CompletableActorFuture;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import org.agrona.DirectBuffer;
+
+@SuppressWarnings({"rawtypes", "unchecked"})
+public class BpmnStepProcessor implements TypedRecordProcessor<WorkflowInstanceRecord> {
+
+  private final WorkflowInstanceIndex workflowInstanceIndex;
+  private final WorkflowCache workflowCache;
+  private final BpmnStepContext context = new BpmnStepContext<>();
+  private final Map<BpmnStep, BpmnStepHandler> stepHandlers = new EnumMap<>(BpmnStep.class);
+
+  private ActorControl actor;
+
+  public BpmnStepProcessor(
+      WorkflowCache workflowCache,
+      WorkflowInstanceIndex workflowInstanceIndex,
+      ActivityInstanceMap activityInstanceMap,
+      PayloadCache payloadCache) {
+
+    this.workflowInstanceIndex = workflowInstanceIndex;
+    this.workflowCache = workflowCache;
+
+    // activity
+    stepHandlers.put(BpmnStep.CREATE_JOB, new CreateJobHandler());
+    stepHandlers.put(
+        BpmnStep.APPLY_INPUT_MAPPING,
+        new InputMappingHandler(payloadCache, workflowInstanceIndex, activityInstanceMap));
+    stepHandlers.put(
+        BpmnStep.APPLY_OUTPUT_MAPPING,
+        new OutputMappingHandler(payloadCache, workflowInstanceIndex, activityInstanceMap));
```

Comment:
If we have the handle method directly in the step subclass we also have not to do this mapping.

Reply:
This is makes it also easier in the future to add more stuff